src/containers/ParticleContainer.h

```diff
@@ -5,26 +5,25 @@
  *      Author: tchipevn
  */
 
-#ifndef DEPENDENCIES_EXTERNAL_AUTOPAS_SRC_PARTICLECONTAINER_H_
-#define DEPENDENCIES_EXTERNAL_AUTOPAS_SRC_PARTICLECONTAINER_H_
+#pragma once
 
 #include <array>
+#include "ParticleContainerInterface.h"
 #include "iterators/ParticleIterator.h"
 #include "iterators/RegionParticleIterator.h"
 #include "pairwiseFunctors/Functor.h"
 
 namespace autopas {
 
-// consider multiple inheritance or delegation vor avoidane of virtual call to
-// Functor
+// consider multiple inheritance or delegation to avoid virtual call to Functor
 /**
  * The ParticleContainer class stores particles in some object and provides
  * methods to iterate over its particles.
  * @tparam Particle Class for particles
  * @tparam ParticleCell Class for the particle cells
  */
 template <class Particle, class ParticleCell>
-class ParticleContainer {
+class ParticleContainer : public ParticleContainerInterface<Particle> {
```

Comment:
when compiling with g++7 I'm still getting lots of warinings that methods can be marked as override here (mostly for getters & setters). Apparently you need to put both `final` and `override` (order irrelevant but plz consistent ;) ).

Reply:
sounds like a bug in g++7
final implicitly always means override

Reply:
ok, added the overrides, even though clang-tidy complains about this.
(we should probably add a .clang-tidy file + add this to jenkins...)